extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessor.java

```diff
@@ -0,0 +1,127 @@
+import io.deephaven.qst.type.BooleanType;
+import io.deephaven.qst.type.BoxedType;
+import io.deephaven.qst.type.ByteType;
+import io.deephaven.qst.type.CharType;
+import io.deephaven.qst.type.DoubleType;
+import io.deephaven.qst.type.FloatType;
+import io.deephaven.qst.type.GenericType;
+import io.deephaven.qst.type.InstantType;
+import io.deephaven.qst.type.IntType;
+import io.deephaven.qst.type.LongType;
+import io.deephaven.qst.type.ShortType;
+import io.deephaven.qst.type.Type;
+
+import java.time.Instant;
+import java.util.List;
+
+/**
+ * An interface for processing data from one or more input objects into output chunks on a 1-to-1 input to output basis.
+ *
+ * @param <T> the object type
+ * @see ObjectProcessorRowLimited
+ */
+public interface ObjectProcessor<T> {
+
+    /**
+     * Creates or returns an implementation that adds strict safety checks around {@code delegate}
+     * {@link #processAll(ObjectChunk, List)}. The may be useful during for development or debugging purposes.
+     *
+     * @param delegate the delegate
+     * @return the strict implementation
+     * @param <T> the object type
+     */
+    static <T> ObjectProcessor<T> strict(ObjectProcessor<T> delegate) {
+        return ObjectProcessorStrict.create(delegate);
+    }
+
+    /**
+     * The relationship between {@link #outputTypes() output types} and the {@link #processAll(ObjectChunk, List)
+     * processAll out param} {@link WritableChunk#getChunkType()}.
+     *
+     * <table>
+     * <tr>
+     * <th>{@link Type}</th>
+     * <th>{@link ChunkType}</th>
+     * </tr>
+     * <tr>
+     * <td>{@link ByteType}</td>
+     * <td>{@link ChunkType#Byte}</td>
+     * </tr>
+     * <tr>
+     * <td>{@link ShortType}</td>
+     * <td>{@link ChunkType#Short}</td>
+     * </tr>
+     * <tr>
+     * <td>{@link IntType}</td>
+     * <td>{@link ChunkType#Int}</td>
+     * </tr>
+     * <tr>
+     * <td>{@link LongType}</td>
+     * <td>{@link ChunkType#Long}</td>
+     * </tr>
+     * <tr>
+     * <td>{@link FloatType}</td>
+     * <td>{@link ChunkType#Float}</td>
+     * </tr>
+     * <tr>
+     * <td>{@link DoubleType}</td>
+     * <td>{@link ChunkType#Double}</td>
+     * </tr>
+     * <tr>
+     * <td>{@link CharType}</td>
+     * <td>{@link ChunkType#Char}</td>
+     * </tr>
+     * <tr>
+     * <td>{@link BooleanType}</td>
+     * <td>{@link ChunkType#Byte} (<b>not</b> {@link ChunkType#Boolean})</td>
+     * </tr>
+     * <tr>
+     * <td>{@link BoxedType}</td>
+     * <td>Same as {@link BoxedType#primitiveType()} would yield.</td>
+     * </tr>
+     * <tr>
+     * <td>{@link InstantType}</td>
+     * <td>{@link ChunkType#Long} ({@link io.deephaven.time.DateTimeUtils#epochNanos(Instant)})</td>
+     * </tr>
+     * <tr>
+     * <td>All other {@link GenericType}</td>
+     * <td>{@link ChunkType#Object}</td>
+     * </tr>
+     * </table>
+     */
+    static ChunkType chunkType(Type<?> type) {
+        return ObjectProcessorTypes.of(type);
+    }
+
+    /**
```

Comment:
The types in TableDefinition are

```
  Class<TYPE> dataType,
  Class<?> componentType
```

- `Type` is our new canonical class for representing that information. I would very much like to change TableDefinition impl to use `Type<TYPE>`. Given reasons already mentioned, I don't think it's appropriate to have column name associations in the ObjectProcessor interface, so that's the reason for not using `TableDefinition`.